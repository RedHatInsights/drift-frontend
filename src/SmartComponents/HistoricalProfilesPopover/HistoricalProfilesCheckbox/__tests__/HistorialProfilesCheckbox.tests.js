import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';

import ConnectedHistoricalProfilesCheckbox, { HistoricalProfilesCheckbox } from '../HistoricalProfilesCheckbox';

/*eslint-disable camelcase*/
describe('HistoricalProfilesCheckbox', () => {
    let props;

    beforeEach(() => {
        props = {
            selectedHSPIds: [],
            profile: { id: '1234', captured_date: '03 May 2020, 18:20 UTC' }
        };
    });

    it('should render correctly', () => {
        const wrapper = shallow(
            <HistoricalProfilesCheckbox { ...props }/>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render mount correctly', () => {
        const wrapper = mount(
            <HistoricalProfilesCheckbox { ...props }/>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe('ConnectedHistoricalProfilesCheckbox', () => {
    let initialState;
    let props;
    let mockStore;

    beforeEach(() => {
        mockStore = configureStore();
        initialState = {
            entities: {}
        };

        props = {
            profile: { id: '1234', captured_date: '03 May 2020, 18:20 UTC' },
            selectedHSPIds: []
        };
    });

    it('should render correctly', () => {
        const store = mockStore(initialState);
        const wrapper = mount(
            <MemoryRouter keyLength={ 0 }>
                <Provider store={ store }>
                    <ConnectedHistoricalProfilesCheckbox { ...props } />
                </Provider>
            </MemoryRouter>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
/*eslint-enable camelcase*/
