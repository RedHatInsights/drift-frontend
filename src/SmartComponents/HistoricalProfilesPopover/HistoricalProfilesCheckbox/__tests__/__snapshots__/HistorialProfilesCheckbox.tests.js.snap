// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedHistoricalProfilesCheckbox should render correctly 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Connect(HistoricalProfilesCheckbox)
        profile={
          Object {
            "captured_date": "03 May 2020, 18:20 UTC",
            "id": "1234",
          }
        }
        selectedHSPIds={Array []}
      >
        <HistoricalProfilesCheckbox
          dispatch={[Function]}
          entities={Object {}}
          profile={
            Object {
              "captured_date": "03 May 2020, 18:20 UTC",
              "id": "1234",
            }
          }
          selectedHSPIds={Array []}
        >
          <Checkbox
            aria-label="1234"
            className=""
            data-ouia-component-id="hsp-popover-option-checkbox-1234"
            data-ouia-component-type="PF4/Checkbox"
            id="1234"
            isChecked={false}
            isDisabled={false}
            isValid={true}
            label="03 May 2020, 18:20 UTC"
            name="1234"
            onChange={[Function]}
          >
            <div
              className="pf-c-check"
            >
              <input
                aria-invalid={false}
                aria-label="1234"
                checked={false}
                className="pf-c-check__input"
                data-ouia-component-id="hsp-popover-option-checkbox-1234"
                data-ouia-component-type="PF4/Checkbox"
                disabled={false}
                id="1234"
                name="1234"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="pf-c-check__label"
                htmlFor="1234"
              >
                03 May 2020, 18:20 UTC
              </label>
            </div>
          </Checkbox>
        </HistoricalProfilesCheckbox>
      </Connect(HistoricalProfilesCheckbox)>
    </Provider>
  </Router>
</MemoryRouter>
`;

exports[`HistoricalProfilesCheckbox should render correctly 1`] = `
<Fragment>
  <Checkbox
    aria-label="1234"
    className=""
    data-ouia-component-id="hsp-popover-option-checkbox-1234"
    data-ouia-component-type="PF4/Checkbox"
    id="1234"
    isChecked={false}
    isDisabled={false}
    isValid={true}
    label="03 May 2020, 18:20 UTC"
    name="1234"
    onChange={[Function]}
  />
</Fragment>
`;

exports[`HistoricalProfilesCheckbox should render mount correctly 1`] = `
<HistoricalProfilesCheckbox
  profile={
    Object {
      "captured_date": "03 May 2020, 18:20 UTC",
      "id": "1234",
    }
  }
  selectedHSPIds={Array []}
>
  <Checkbox
    aria-label="1234"
    className=""
    data-ouia-component-id="hsp-popover-option-checkbox-1234"
    data-ouia-component-type="PF4/Checkbox"
    id="1234"
    isChecked={false}
    isDisabled={false}
    isValid={true}
    label="03 May 2020, 18:20 UTC"
    name="1234"
    onChange={[Function]}
  >
    <div
      className="pf-c-check"
    >
      <input
        aria-invalid={false}
        aria-label="1234"
        checked={false}
        className="pf-c-check__input"
        data-ouia-component-id="hsp-popover-option-checkbox-1234"
        data-ouia-component-type="PF4/Checkbox"
        disabled={false}
        id="1234"
        name="1234"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="pf-c-check__label"
        htmlFor="1234"
      >
        03 May 2020, 18:20 UTC
      </label>
    </div>
  </Checkbox>
</HistoricalProfilesCheckbox>
`;
